### PR TITLE
[BBC-8375] Remove default exports

### PR DESCRIPTION
**Pull request by mcellier**
## Description

Remove all default exports, to match our guidelines. 

## What has been done

- Remove default exports (components & reexports)
- Added eslint rule `import/no-default-export`
- Updated component template

## Things to consider

N/A

## How it was tested

Storybook + kairos

### CODE REVIEW

File: src/_internals/card/index.tsx [hunk 1]
```diff
@@ -18,4 +18,4 @@ const StyledCard = styled(Card)`
 `
 
 export { StyledCard as Card }
-export default StyledCard
+StyledCard
```
Comment (wakooka) [inline src/_internals/card/index.tsx:21]: Remove?

Comment (mcellier) [reply src/_internals/card/index.tsx:21]: WTH

File: src/typography/index.tsx [hunk 2]
```diff
@@ -10,4 +10,4 @@ const StyledText = styled(Text)`
 `
 
 export { StyledText as Text }
-export default StyledText
+StyledText
```
Comment (wakooka) [inline src/typography/index.tsx:13]: Coucou

File: src/itemData/ItemData.tsx [hunk 3]
```diff
@@ -55,4 +55,4 @@ export const ItemData = (props: ItemDataProps) => {
   )
 }
 
-export default ItemData
+ItemData
```
Comment (wakooka) [inline src/itemData/ItemData.tsx:58]: 👋

**Review by wakooka — APPROVED** [review]

**Review by simonrelet — APPROVED** [review]
👏👏👏

**Review by Tuxosaurus — APPROVED** [review]

**Comment by gabrielagabriel** [comment]
Amazing, are you taking caring of applying the breaking change on the App!?

**Comment by mcellier** [comment]
> Amazing, are you taking caring of applying the breaking change on the App!?

PR is already merged on kairos.

**Pull request merged**